internal/paascharm/gostructs_test.go

```diff
@@ -0,0 +1,162 @@
+		})
+	}
+}
+
+func TestIntegrations(t *testing.T) {
+	var tests = []struct {
+		name           string
+		charmcraftYaml paascharm.CharmcraftYamlConfig
+		expected       paascharm.GoStructsData
+	}{
+		{
+			"mongodb mysql redis and postgresql",
+			paascharm.CharmcraftYamlConfig{
+				Requires: map[string]paascharm.CharmcraftIntegration{
+					"mongodb":    {Interface: "mongodb_client"},
+					"mysql":      {Interface: "mysql_client"},
+					"postgresql": {Interface: "postgresql_client"},
+					"redis":      {Interface: "redis"},
+				},
+			},
+			paascharm.GoStructsData{
+				PackageName:             "pkg",
+				CommonPrefix:            "APP_",
+				HasDatabaseIntegrations: true,
+				Integrations: map[string]paascharm.Integration{
+					"mongodb":    {Name: "mongodb", GoName: "MongoDB", Interface: "mongodb_client", IsDatabase: true, DatabasePrefix: "APP_MONGODB_"},
+					"mysql":      {Name: "mysql", GoName: "MySQL", Interface: "mysql_client", IsDatabase: true, DatabasePrefix: "APP_MYSQL_"},
+					"postgresql": {Name: "postgresql", GoName: "PostgreSQL", Interface: "postgresql_client", IsDatabase: true, DatabasePrefix: "APP_POSTGRESQL_"},
+					"redis":      {Name: "redis", GoName: "Redis", Interface: "redis", IsDatabase: true, DatabasePrefix: "APP_REDIS_"},
+				},
+			},
+		},
+		{
+			"optional s3 and saml, no database integration",
+			paascharm.CharmcraftYamlConfig{
+				Requires: map[string]paascharm.CharmcraftIntegration{
+					"s3":   {Interface: "s3", Optional: true},
+					"saml": {Interface: "saml"},
+				},
+			},
+			paascharm.GoStructsData{
+				PackageName:  "pkg",
+				CommonPrefix: "APP_",
+				Integrations: map[string]paascharm.Integration{
+					"s3":   {Name: "s3", GoName: "S3", Interface: "s3", Optional: true},
+					"saml": {Name: "saml", GoName: "SAML", Interface: "saml"},
+				},
+			},
+		},
+		{
+			"unknown integration",
+			paascharm.CharmcraftYamlConfig{
+				Requires: map[string]paascharm.CharmcraftIntegration{
+					"unknown": paascharm.CharmcraftIntegration{Interface: "Unknown"},
+				},
+			},
+			paascharm.GoStructsData{PackageName: "pkg", CommonPrefix: "APP_"},
+		},
+	}
+	for _, tt := range tests {
+		testname := fmt.Sprintf("%s", tt.name)
+		t.Run(testname, func(t *testing.T) {
+			actual, err := paascharm.NewGoStructsData("pkg", tt.charmcraftYaml)
+			if err != nil {
+				t.Fatalf("Error creating Go structs data %s", err)
+			}
+
+			if diff := pretty.Diff(actual, tt.expected); len(diff) > 0 {
+				t.Errorf("go structs data is not correctly generated. diff: %s\n", diff)
+			}
+		})
+	}
+}
+
+// Config options is a map in yaml and a slice in GoStructsData. Order the slice by GoVarName
+// so two GoStructsData can be easily compared
+func normalizeGoStructsData(goStructsData paascharm.GoStructsData) paascharm.GoStructsData {
+	result := goStructsData
+	orderedOptions := make([]paascharm.Option, len(result.Options))
+	copy(orderedOptions, result.Options)
+	sort.Slice(result.Options, func(i, j int) bool {
```

Comment:
Would it be better just to sort after parsing? That puts it in a consistent order for all users, not just tests, and would simplify the test.